packages/teamscale-javascript-instrumenter/src/instrumenter/Postprocessor.ts

```diff
@@ -0,0 +1,364 @@
+import { parse } from '@babel/parser';
+import generate from '@babel/generator';
+import traverse, { NodePath } from '@babel/traverse';
+import {
+	Node,
+	CallExpression,
+	Identifier,
+	isCallExpression,
+	isIdentifier,
+	isMemberExpression,
+	NumericLiteral,
+	SourceLocation,
+	UpdateExpression,
+	StringLiteral,
+	isVariableDeclaration,
+	VariableDeclaration,
+	VariableDeclarator,
+	isFunctionDeclaration,
+	ExpressionStatement,
+	isSequenceExpression,
+	SequenceExpression,
+	isUpdateExpression
+} from '@babel/types';
+import { IllegalStateException } from '@cqse/commons';
+
+const COVERAGE_OBJ_FUNCTION_NAME_PREFIX = 'cov_';
+
+type CoverageIncrement = {
+	type: 'statement' | 'branch' | 'function';
+	coverageObjectId: string;
+};
+
+type StatementCoverageIncrement = CoverageIncrement & {
+	type: 'statement';
+	statementId: number;
+};
+
+type BranchCoverageIncrement = CoverageIncrement & {
+	type: 'branch';
+	branchId: number;
+	locationId: number;
+};
+
+type FunctionCoverageIncrement = CoverageIncrement & {
+	type: 'function';
+	functionId: number;
+};
+
+function isIstanbulCoverageFunctionDeclaration(node: Node | undefined): string | undefined {
+	if (!isFunctionDeclaration(node)) {
+		return undefined;
+	}
+
+	const functionName = (node.id as Identifier).name;
```

Comment:
Gets rid of the the forced type cast and also safely deals with id not being defined
```suggestion
	const functionName = node.id?.name;
```